### PR TITLE
linux: musl libc

### PR DESCRIPTION
**Pull request by aheadmode**
Signed-off-by: tae8aed0aichuofe <uziethiuth2upaet@protonmail.com>

Commit Message: linux: undefine pthread_getname_np for musl libc
Additional Description: pthread_getname_np is not available in musl libc
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

CC @htuch @jaychenatr

### CODE REVIEW

**Review by antoniovicente — COMMENTED** [review]
Thanks for the build fix.  Are there plans to introduce a bazel config to enable building against musl or instructions on how to enable use of musl libc?

**Comment by aheadmode** [comment]
Yes, I am actively working on that.

Obvious compiler fixes will be pushed upstream. The final Starlark rules for the C++ musl toolchain will come in one batch. 
At some point I might need help from the community. 

```
$ bazel build --config:musl //...
```

**Comment by aheadmode** [comment]
https://github.com/envoyproxy/envoy/issues/13998

File: include/envoy/common/platform.h [hunk 1]
```diff
@@ -291,8 +291,10 @@ struct mmsghdr {
 #define SUPPORTS_PTHREAD_NAMING 1
 #endif // __ANDROID_API__ >= 26
 #elif defined(__linux__)
+#if defined(__GLIBC__)
 #undef SUPPORTS_PTHREAD_NAMING
 #define SUPPORTS_PTHREAD_NAMING 1
+#endif // defined(__GLIBC__)
```
Comment (antoniovicente) [inline include/envoy/common/platform.h:297]: Seems that the format check is failing.  Please add GLIBC to tools/spelling/spelling_dictionary.txt

**Pull request closed without merging**